resteasy-core/src/main/java/org/jboss/resteasy/core/StringParameterInjector.java

```diff
@@ -755,11 +758,19 @@ else if (valueOf != null)
       }
       catch (Exception e)
       {
-         throwProcessingException(Messages.MESSAGES.unableToExtractParameter(getParamSignature(), strVal, target), e);
+         throwProcessingException(Messages.MESSAGES.unableToExtractParameter(getParamSignature(), _encode(strVal), target), e);
       }
       return null;
    }
 
+   private String _encode(String strVal) {
+      try {
+         return URLEncoder.encode(strVal, StandardCharsets.UTF_8.toString());
+      } catch (UnsupportedEncodingException e) {
+         return e.getMessage();

```

Comment:
Should the message be returned here or the original value?

Reply:
Perhaps it would be better to `log.debug()` this exception and return the original value

Reply:
@gastaldi the original value is unescaped and still unsafe right?

Reply:
adding a debug log is fine, let me add it in this PR. For return value, I'm not sure what to return for an exception, maybe assume this won't happen?

Reply:
> @gastaldi the original value is unescaped and still unsafe right?

Yes. I just mentioned that because `e.getMessage()` will return `UTF-8`

Reply:
Perhaps in this case the best is to just rethrow the exception as a RuntimeException